src/routes/courierPage.tsx

```diff
@@ -1,314 +1,314 @@
-      return;
-    }
-
-    try {
-      const deliveryCollection = collection(db, "delivery");
-      const vaultCollection = collection(db, "vault");
-
-      // Query the vault collection for the selected compartment
-      const vaultQuery = query(
-        vaultCollection,
-        where("compartment", "==", isNaN(Number(selectedCompartment))
-          ? selectedCompartment // Treat as string if not a valid number
-          : Number(selectedCompartment) // Treat as number if valid
-        )
-      );
-      const vaultSnapshot = await getDocs(vaultQuery);
-
-      if (!vaultSnapshot.empty) {
-        const vaultDocId = vaultSnapshot.docs[0].id;
-        const vaultDocRef = doc(db, "vault", vaultDocId);
-
-        // Update the status of the compartment to "blocked"
-        await updateDoc(vaultDocRef, { status: "blocked" });
-
-        console.log(
-          `Compartment ${selectedCompartment} status updated to blocked in the vault table.`
-        );
-      } else {
-        console.error(
-          `Compartment ${selectedCompartment} not found in the vault table.`
-        );
-        Swal.fire({
-          icon: "error",
-          title: "Error",
-          text: `Compartment ${selectedCompartment} not found in the vault table.`,
-        });
-        return;
-      }
-
-      // Save the data to the delivery table
-      const deliveryData = {
-        user: selectedUser.value,
-        packsize: selectedSize.value,
-        compartment: selectedCompartment,
-        courier: selectedCourier,
-        status: "pending", // Default status
-        timestamp: Timestamp.now(), // Use imported Timestamp
-      };
-
-      console.log("Attempting to save delivery:", deliveryData);
-
-      await addDoc(deliveryCollection, deliveryData);
-
-      Swal.fire({
-        icon: "success",
-        title: "Success",
-        text: "Delivery details saved successfully, and compartment status updated!",
-      }).then(() => {
-        // Refresh the page on success
-        window.location.reload();
-      });
-
-      // Reset form (not necessary with page refresh but added for completeness)
-      setSelectedUser(null);
-      setSelectedSize(null);
-      setSelectedCompartment("");
-      setSelectedCourier("Amazon");
-    } catch (error) {
-      console.error("Error during submission:", error);
-      Swal.fire({
-        icon: "error",
-        title: "Error",
-        text: "Failed to save delivery details or update compartment status. Please try again later.",
-      });
-    }
-  };
-
-  return (
-    <>
-      {/* Header Area */}
-      <div className="header-area" id="headerArea">
-        <div className="container">
-          <div className="header-content position-relative d-flex align-items-center justify-content-between">
-            {/* Back Button */}
-            <div className="back-button">
-              <Link to="/">
-                <i className="bi bi-arrow-left-short"></i>
-              </Link>
-            </div>
```

Comment:
This button has been used somewhere else, it might be a good candidate for its own component.